rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/locations/impl/TreeSearch.java

```diff
@@ -0,0 +1,126 @@
+package org.rascalmpl.vscode.lsp.util.locations.impl;
+
+import org.rascalmpl.values.IRascalValueFactory;
+import org.rascalmpl.values.parsetrees.ITree;
+import org.rascalmpl.values.parsetrees.TreeAdapter;
+
+import io.usethesource.vallang.IList;
+import io.usethesource.vallang.IListWriter;
+import io.usethesource.vallang.ISourceLocation;
+import io.usethesource.vallang.IValue;
+
+/**
+ * Utilities for finding sub-trees based on UTF-32 line/column indexing.
+ */
+public class TreeSearch {
+
+    /**
+     * Helper to test if a line/column position is within the scope of a source location's
+     * begin line/column and end line/column scope. The first and the last lines are
+     * where the column information is used.
+     *
+     * @param loc     location with line/column information that acts as a range
+     * @param line    line to check if its within the range
+     * @param column  column to see if we are within the range
+     * @return true iff the given `loc` range spans around the line/column position, inclusively.
+     */
+    private static boolean inside(ISourceLocation loc, int line, int column) {
+        line += 1; // LSP line offsets are 0-based, while Rascal tree locations are 1-based.
```

Comment:
This should be done outside of this class. There we can handle the mapping from LSP line and columns to Rascal lines and columns. This class should be only in the Rascal world (now lines are mapped but columns are still not.)

Reply:
I just removed the fix here. In the surrounding API there was already a correction for 1-based to 0-based. So it is strange that this worked, and it's strange that it still works.